### PR TITLE
Node: Add test module to v16

### PR DESCRIPTION
**Pull request by johan13**
This MR copies test.d.ts mostly unmodified from v18 to v16.

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V16.md#experimental-test-runner
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/node/v16/test.d.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
+/**
+ * The `node:test` module provides a standalone testing module.
+ * @see [source](https://github.com/nodejs/node/blob/v18.0.0/lib/test.js)
```
Comment (peterblazejewicz) [inline types/node/v16/test.d.ts:3]: this is obviously out-of-the-context, there should be `v16.*.*` in url, like e.g.:
https://github.com/nodejs/node/blob/v16.17.0/lib/test.js
(should be one feature was introduced)

Comment (peterblazejewicz) [reply types/node/v16/test.d.ts:3]: The same for:
```ts
@since v18.6.0
```

**Comment by textbook** [comment]
Related: #61837. Whichever is merged first, the other needs to also reflect.

**Comment by johan13** [comment]
> Related: #61837. Whichever is merged first, the other needs to also reflect.

Thanks. I have copied textbook's v18 changes to v16.

**Comment by peterblazejewicz** [comment]
@johan13 everything's great. Can you include/review changes from this just merged PR?
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61837/files

**Comment by johan13** [comment]
> @johan13 everything's great. Can you include/review changes from this just merged PR?
> https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61837/files

Of course. Those changes are already copied into this PR.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by peterblazejewicz — APPROVED** [review]

**Comment by johan13** [comment]
Ready to merge

**Pull request merged**